packages/flutter_tools/templates/app_shared/windows.tmpl/runner/utils.cpp

```diff
@@ -45,17 +45,18 @@ std::string Utf8FromUtf16(const wchar_t* utf16_string) {
   if (utf16_string == nullptr) {
     return std::string();
   }
+  int source_length = static_cast<int>(wcslen(utf16_string));
   int target_length = ::WideCharToMultiByte(
       CP_UTF8, WC_ERR_INVALID_CHARS, utf16_string,
-      -1, nullptr, 0, nullptr, nullptr);
+      source_length, nullptr, 0, nullptr, nullptr);
   std::string utf8_string;
   if (target_length == 0 || target_length > utf8_string.max_size()) {
     return utf8_string;
   }
   utf8_string.resize(target_length);
   int converted_length = ::WideCharToMultiByte(
       CP_UTF8, WC_ERR_INVALID_CHARS, utf16_string,
-      -1, utf8_string.data(),
+      source_length, utf8_string.data(),
       target_length, nullptr, nullptr);
   if (converted_length == 0) {
```

Comment:
Is this case possible?

Reply:
Well, if we know the lengths of both the source string and resulting string we should pass them. But I'll change this to -1 and subtract 1 from target_length anyway as per the other comment above.

Reply:
See my comment above, you shouldn't need to subtract one. The output size is the number of bytes required for the buffer.

> Is this case possible?

It seems unlikely given that we just allocated a buffer the right size, but without knowing the implementation, checking the return value here seems like the right thing to do and it's an O(1) check, so pretty negligible compared to the encoding conversion that `WideCharToMultibyte` is doing.